libs/dh/metering-point/feature-upload-measurements/src/upload-service.ts

```diff
@@ -0,0 +1,97 @@
+//#region License
+/**
+ * @license
+ * Copyright 2020 Energinet DataHub A/S
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License2");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+//#endregion
+import { effect, Injectable } from '@angular/core';
+import {
+  ElectricityMarketMeteringPointType,
+  MeteringPointType2,
+  SendMeasurementsDocument,
+  SendMeasurementsResolution,
+} from '@energinet-datahub/dh/shared/domain/graphql';
+import { MeasureDataResult } from './models/measure-data-result';
+import { injectToast } from '@energinet-datahub/dh/shared/ui-util';
+import { mutation } from '@energinet-datahub/dh/shared/util-apollo';
+import { assert, assertIsDefined } from '@energinet-datahub/dh/shared/util-assert';
+import { parseMeasurements } from './parse-measurements';
+
+@Injectable({ providedIn: 'root' })
+export class DhUploadMeasurementsService {
+  private toast = injectToast('meteringPoint.measurements.upload.toast');
```

Comment:
Nitpick: If i understand i correctly when according to https://angular.dev/style-guide#use-readonly-on-properties-that-are-initialized-by-angular it should be private readonly? :-)